### DISCUSSION TITLE
How to print the raw request

### DISCUSSION THREAD

**Discussion by baconcheese113** [heftapp/graphql_codegen#162] [Q&A] [answered]
When using fragments it can be time consuming to manually stitch together the query being sent for debugging purposes. Is there a way to print it?

**Comment by budde377** [comment]
Under the hood we're using `gql`, so I'm sure we can use that to print the document. Regarding stitching, are you thinking of anything other than printing the document sent?

**Reply by baconcheese113** [reply]
For my use-case just printing the query stitched together so that I can copy it into Apollo Studio and debug why it's failing against the public graphql server in my example app. I don't know if there is any demand for a feature like this, so even if there was some manual workaround I'd be happy with it. There aren't any Input params to worry about either

**Comment by budde377** [comment, marked as answer]
I haven't tested this, but I recon you can do something like

```dart
import 'package:gql/language.dart';

main() {
  print(printNode(yourDocumentNodeGeneratedByGraphqlCodegen));
}

```

**Reply by baconcheese113** [reply]
Would `yourDocumentNode` be one of the generated Query classes like `Query$MyQuery$Widget`?

**Reply by budde377** [reply]
It'd be the generated document node, e.g. https://github.com/heftapp/graphql_codegen/blob/main/packages/graphql_codegen/test/assets/add_typenames/schema.graphql.dart#L439
